src/test/java/net/snowflake/client/jdbc/BindingDataLatestIT.java

```diff
@@ -122,57 +126,186 @@ public void testTimestampBindingWithNTZType() throws SQLException {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Tokyo"));
+        Timestamp currT = new Timestamp(System.currentTimeMillis());
+
+        // insert using regular binging
+        executePsStatementForTimestampTest(connection, "regularinsert", currT);
+
+        // insert using stage binding
+        statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
+        executePsStatementForTimestampTest(connection, "stageinsert", currT);
+
+        // Compare the results
+        try (ResultSet rs1 = statement.executeQuery("select * from stageinsert");
+            ResultSet rs2 = statement.executeQuery("select * from regularinsert")) {
+          assertTrue(rs1.next());
+          assertTrue(rs2.next());
+
+          assertEquals(rs1.getInt(1), rs2.getInt(1));
+
+          // Check that all the values are the same.
+          assertEquals(rs1.getTimestamp(2), rs1.getTimestamp(3));
+          assertEquals(rs1.getTimestamp(3), rs1.getTimestamp(4));
+
+          assertEquals(rs1.getTimestamp(2), rs2.getTimestamp(2));
+          assertEquals(rs1.getTimestamp(3), rs2.getTimestamp(3));
+          assertEquals(rs1.getTimestamp(4), rs2.getTimestamp(4));
+        }
+      } finally {
+        statement.execute("drop table if exists stageinsert");
+        statement.execute("drop table if exists regularinsert");
+        TimeZone.setDefault(origTz);
+      }
+    }
+  }
+
+  /**
+   * Test that stage binding and regular binding insert and return the same value for timestamp_ltz
+   * when the local timezone has the daylight saving. This test is added in version > 3.16.1
+   *
+   * <p>When CLIENT_TIMESTAMP_TYPE_MAPPING setting is mismatched with target data type (e.g
+   * MAPPING=LTZ and insert to NTZ or MAPPING=NTZ and insert to TZ/LTZ there could be different
+   * result as the timezone offset is applied on client side and removed on server side. This only
+   * occurs around the boundary of daylight-savings and the difference from the source data would be
+   * one hour. Both regular binding and stage binding have such issue but they also behave
+   * diffently, for some data only regular binding gets the extra hour while sometime only stage
+   * binding does. The workaround is to use CLIENT_TIMESTAMP_TYPE_MAPPING=LTZ to insert LTZ/TZ data
+   * and use CLIENT_TIMESTAMP_TYPE_MAPPING=NTZ to insert NTZ data.
+   *
+   * <p>This test cannot run on the GitHub testing because of the "ALTER SESSION SET
+   * CLIENT_STAGE_ARRAY_BINDING_THRESHOLD" This command should be executed with the system admin.
+   *
+   * @throws SQLException
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testTimestampBindingWithLTZTypeForDayLightSavingTimeZone() throws SQLException {
+    TimeZone origTz = TimeZone.getDefault();
+    Calendar australia = Calendar.getInstance(TimeZone.getTimeZone("Australia/Sydney"));
+    TimeZone.setDefault(TimeZone.getTimeZone("Australia/Sydney"));
```

Comment:
let's extract the tested TimeZone to variable

Reply:
Done.